libsupport/include/katana/Random.h

```diff
@@ -8,15 +8,26 @@
 
 namespace katana {
 
-KATANA_EXPORT std::string RandomAlphanumericString(uint64_t len);
-// Range 0..len-1, inclusive
-KATANA_EXPORT int64_t RandomUniformInt(int64_t len);
-// Range min+1..max-1, inclusive
-KATANA_EXPORT int64_t RandomUniformInt(int64_t min, int64_t max);
-// Range 0.0f..max, inclusive
-KATANA_EXPORT float RandomUniformFloat(float max);
-
-KATANA_EXPORT std::mt19937& GetGenerator();
+using RandGenerator = std::mt19937;
+
+/// Generate a random alphanumeric string of length \param len using
+/// \param gen if provided. If no generator is specified, uses a generator
+/// seeded by the OS's randomness
+KATANA_EXPORT std::string RandomAlphanumericString(
+    uint64_t len, RandGenerator* gen = nullptr);
+
+/// Range 0..len-1, inclusive
+KATANA_EXPORT int64_t
+RandomUniformInt(int64_t len, RandGenerator* gen = nullptr);
+
+/// Range min+1..max-1, inclusive
```

Comment:
fwiw; this is confusing to me since it opens up corner cases like `min = max - 1` and the max is inclusive for floats; could we make them all inclusive? (but then they behave just like `std::uniform_int_distribution` so maybe the real answer is to ditch these?

Reply:
I don't think I knew about `std::uniform_int_distribution` when I wrote this.  That looks fine.